lib/services/restaurant_search_api.dart

```diff
@@ -1,27 +1,140 @@
+      Uri.parse(
+          "https://naveropenapi.apigw.ntruss.com/map-reversegeocode/v2/gc?request=coordsToaddr&coords=$lon,$lat&sourcecrs=epsg:4326&output=json&orders=addr,roadaddr"),
+      headers: headerss);
+  String jsonData = response.body;
+  Map<String, dynamic> userAddress = jsonDecode(jsonData);
+  var address = Address.fromJson(userAddress);
+  String userLocation = address.toString();
+  return userLocation;
+}
+
+class Address {
+  final String si;
+  final String gu;
+  final String dong;
+  final String detail1;
+  final String detail2;
+
+  Address(this.si, this.gu, this.dong, this.detail1, this.detail2);
+
+  Address.fromJson(Map<String, dynamic> json)
+      : si = json["results"][0]['region']['area1']['name'],
+        gu = json["results"][0]['region']['area2']['name'],
+        dong = json["results"][0]['region']['area3']['name'],
+        detail1 = json["results"][0]['land']['number1'],
+        detail2 = json["results"][0]['land']['number2'];
+
+  Map<String, dynamic> toJson() => {
+        'si': si,
+        'gu': gu,
+        'dong': dong,
+        'detail1': detail1,
+        'detail2': detail2,
+      };
+
+  @override
+  String toString() {
+    return si + " " + gu + " " + dong + " " + detail1 + "-" + detail2;
+  }
+}
+
+Future<List<RestaurantModel>> searchNearbyRestaurant(
+    String userLocation) async {
+  debugPrint(userLocation);
+  List<RestaurantModel> restaurantList = [];
+
+  for (int i = 0; i < _foodList.length; i++) {
+    final uri = Uri.parse(
+        "https://openapi.naver.com/v1/search/local.json?query=$userLocation ${_foodList[i]}&display=20&start=1&sort=random");
+    final Map<String, String> header = {
+      "X-Naver-Client-Id": "uWDv77C8E2CmY5bUBJ92",
+      "X-Naver-Client-Secret": "aSgxIylTMz",
+      "display": "1"
+    };
+
+    final response = await http.get(uri, headers: header);
+    String jsonRestaurantData = response.body;
+    final restaurantJsonList =
+        (jsonDecode(jsonRestaurantData)["items"] as List<dynamic>);
 
-const String _kNativeAppKey = 'db88ffc1aed093e91db845b457d49495';
-const String _kJavaScriptKey = '7353032fb3c902fbc319097fbc35a44a';
-const String _kRestApiKey = '665888240dc5aa6f10d609f5be6c20d1';
-
-Future<List> searchRestaurant() async {
-  var url = Uri.parse(
-      'https://dapi.kakao.com/v2/maps/sdk.js?appkey=$_kJavaScriptKey&libraries=services');
-  var response = await http.get(url);
-
-  final flutterJs = getJavascriptRuntime();
-  final result = flutterJs.evaluate('''
-script.src = '//dapi.kakao.com/v2/maps/sdk.js?appkey=$_kNativeAppKey&libraries=services&autoload=false';
-script.onload = () => {
-  kakao.maps.load(() => {
-    console.assert(kakao.maps.Map);
-  });
-};
-''');
-  print(result.stringResult);
-  return [];
+    for (final model in restaurantJsonList) {
+      restaurantList
+          .add(RestaurantModel.fromNaverApiJson(model as Map<String, dynamic>));
+    }
+  }
+  List<int> selectedRestaurantList = [];
```

Comment:
변수이름이 헷갈림 아래처럼 하는게 좋을듯 
```suggestion
  List<int> selectedIndexes = [];
```